docs/guide/extending/extending_thumbnails.md

```diff
@@ -0,0 +1,128 @@
+# Extending thumbnails
+
+Thumbnails enable you to easily choose an image for a specific Content Type.
+If you do not want to use custom thumbnails [`ContentType` icons](../../resources/icons.md#content-types) will be used instead.
+
+## Thumbnail mechanism 
+
+The thumbnail mechanism has two layers, and each layer can have many implementations.
+The mechanism checks if any of the implementations returns an image that can be a thumbnail.
+If found, the image is used as a Content Type thumbnail.
+
+First layer of the mechanism contains strategy pattern that focuses on finding a thumbnail.
+The thumbnail can be found inside or outside the Content Type.
+For example for users thumbnails can be downloaded from an avatar-generating service.
+
+For this layer there are two default implementations.
+The first looks for images in Fields, and after that the second one will look for them in the Content Type.
+If there are no Fields that contain an image that can be a thumbnail, the Content Type icon will be used as a thumbnail.
+This mechanism can be modified to fit your site needs, so you can decide from where and how the thumbnails will be downloaded.
+
+Second layer of thumbnail mechanism enables selection of thumbnail from a Field that first layer has found. 
```

Comment:
Is it really a second layer ? Won't the first layer be restricted to fields that have the "Can be a thumbnail" option enabled ? This description feels complex to me. Maybe a schema would help.